src/business-logic/services/device-connection-service-implementation.ts

```diff
@@ -0,0 +1,100 @@
+import { DeviceConnectionService } from './interfaces/device-connection-service'
+import { Device } from '../../model/entities/device'
+import { DeviceConnectionFactory } from './interfaces/device-connection-factory'
+import { DeviceConnection } from './interfaces/inewsgateway-device-connection'
+
+type DeviceAggregate = {
+  deviceConnection: DeviceConnection,
+  device: Device
+}
+
+export class DeviceConnectionServiceImplementation implements DeviceConnectionService {
+  private readonly connectedDevices: Map<string, DeviceAggregate> = new Map()
+
+  constructor(private readonly deviceConnectionFactory: DeviceConnectionFactory) {
+  }
+
+  public async createConnection(device: Device): Promise<DeviceConnectionStatus> {
+    if (this.connectedDevices.has(device.id)) throw new DeviceAlreadyConnectedError(device.id)
+    const deviceConnection: DeviceConnection = this.deviceConnectionFactory.createDeviceConnection(device)
+
+    await deviceConnection.connect()
+
+    this.connectedDevices.set(device.id, {deviceConnection, device})
+
+    return DeviceConnectionStatus.CONNECTED
```

Comment:
You got a point. Due to time constraints I am not gonna refactor that at the moment since it is a matter of preference, but will note the "infer device connection status based on promise resolved/rejected" for the future.

Reply:
(Got a question on this topic further down this page)